### PR TITLE
Groovy sync plugin support files for paremeters

### PR DESCRIPTION
**Pull request by joostoudeman**
Central collecting of properties has been implemented in this branch to avoid these kind of bugs in the future. Also a test for the sync plugin.

### CODE REVIEW

File: bootstrapgenerator/src/main/java/nl/openweb/hippo/groovy/ScriptClassFactory.java [hunk 1]
```diff
@@ -112,10 +113,10 @@ public static ScriptClass getInterpretingClass(final File file, final boolean ke
     public static String readFileEnsuringLinuxLineEnding(final File file) throws IOException {
         String content = FileUtils.fileRead(file);
         if (content.contains(LINE_END_MAC)) {
-            return content.replaceAll(LINE_END_WINDOWS, LINE_END_LINUX)
+            content = content.replaceAll(LINE_END_WINDOWS, LINE_END_LINUX)
                     .replaceAll(LINE_END_MAC, LINE_END_LINUX);
         }
-        return content;
+        return content.replaceAll("[\\t ]+" + NEWLINE, NEWLINE);
```
Comment (mxdrenthe) [inline bootstrapgenerator/src/main/java/nl/openweb/hippo/groovy/ScriptClassFactory.java]: replacing tabs for NewLine? Won't this increase the amount of new lines compared to the original

File: updatersync/src/main/java/nl/openweb/hippo/groovy/GroovyFilesServiceImpl.java [hunk 2]
```diff
@@ -99,25 +100,34 @@ private static boolean setUpdateScriptJcrNode(Node parent, File file) throws Rep
...
+        final Map<String, Object> properties = PropertyCollector.getPropertiesForUpdater(scriptClass, file.getParentFile());
+        for (Map.Entry<String, Object> entry : properties.entrySet()) {
+            if(!JCR_PRIMARY_TYPE.equals(entry.getKey())){
+                scriptNode.setProperty(entry.getKey(), getValue(entry));
+            }
+        }
         return true;
     }
 
+    private static Value getValue(final Map.Entry<String, Object> entry) throws ValueFormatException {
```
Comment (mxdrenthe) [inline updatersync/src/main/java/nl/openweb/hippo/groovy/GroovyFilesServiceImpl.java]: This won't ever be a multi-value field?

File: bootstrapgenerator/src/main/java/nl/openweb/hippo/groovy/XmlGenerator.java [hunk 3]
```diff
@@ -69,29 +62,37 @@ private XmlGenerator() {
...
+            }
+            properties.add(createProperty(entry.getKey(), value, valueType));
         }
-        addStringPropertyIfNotEmpty(properties, HIPPOSYS_QUERY, updater.xpath());
-        addStringPropertyIfNotEmpty(properties, HIPPOSYS_SCRIPT, wrap(script.getContent()));
-        properties.add(createProperty(HIPPOSYS_THROTTLE, updater.throttle(), ValueType.LONG));
         return rootnode;
     }
 
+    private static String getValueType(final Map.Entry<String, Object> entry) {
```
Comment (mxdrenthe) [inline bootstrapgenerator/src/main/java/nl/openweb/hippo/groovy/XmlGenerator.java]: This value will never be an Array?

File: updatersync/src/main/java/nl/openweb/hippo/groovy/GroovyFilesServiceImpl.java (hunk 2 above)
Comment (joostoudeman) [reply updatersync/src/main/java/nl/openweb/hippo/groovy/GroovyFilesServiceImpl.java]: All updater fields seem to be single value. So for now: no.

File: bootstrapgenerator/src/main/java/nl/openweb/hippo/groovy/XmlGenerator.java (hunk 3 above)
Comment (joostoudeman) [reply bootstrapgenerator/src/main/java/nl/openweb/hippo/groovy/XmlGenerator.java]: No, the property can only be of one type.

**Pull request merged**